partner_vat_unique/models/res_partner.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Grant Thornton Spain - Ismael Calvo <ismael.calvo@es.gt.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.constrains('vat')
+    def _check_vat_unique(self):
+        for partner in self:
+            if partner.vat:
+                duplicated_vat = self.search_count([
```

Comment:
@ismaelcj Maybe it could be done with a read_group before the loop as one can experiment performance problems on huge db's (and .

`results = self.env['res.partner'].read_group([('parent_id', '=', False), ('id', 'in', self.ids), ('vat', '!=', False)], ['vat'], ['vat'])
for result in results:
   if result.get('vat_count') > 1:
      ...
`

Reply:
Thanks @rousseldenis I will do the changes